src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -3514,19 +3514,24 @@ def groupby_agg(
         )
 
         # turn each agg function into an AggFuncInfo named tuple, where is_dummy_agg is set to false;
-        # i.e., none of the aggregations here can be dummy.
+        # i.e., none of the aggregations here can be dummy. Also, append the func to agg_col_funcs which
+        # is used to check if the function preserves Snowpark pandas types.
+        agg_col_funcs = []
```

Comment:
You'd better to generate this from `generate_column_agg_info` instead. This block the independency of the function  `convert_func_to_agg_func_info` and also the info you need should be in `column_to_agg_func` already.

Reply:
Do you mean something like this @sfc-gh-azhan:

```
        agg_col_funcs = []
        for agg_col, func in column_to_agg_func.items():
            if is_list_like(func) and not is_named_tuple(func):
                for fn in func:
                    agg_col_funcs.append(fn.func)
            else:
                agg_col_funcs.append(func.func)
```